polyply/tests/test_gen_seq.py

```diff
@@ -0,0 +1,174 @@
+    ref_graph = nx.Graph()
+    ref_graph.add_edges_from(ref_edges)
+    macro = MacroString(macro_str)
+    macro_graph = macro.gen_graph()
+    assert len(nx.get_node_attributes(macro_graph, "resname")) == len(ref_graph.nodes)
+    assert ref_graph.edges == macro_graph.edges
+
+
+def test_generate_seq_graph():
+    macro_strA = "A:6:1:A-1."
+    macro_strB = "B:2:4:B-1."
+
+    macros = {"A": MacroString(macro_strA),
+              "B": MacroString(macro_strB)}
+
+    seq_graph = generate_seq_graph(["A", "B", "A"], macros, ["0:1:2-0", "1:2:1-1"])
+    graphA = macros["A"].gen_graph()
+    graphB = macros["B"].gen_graph()
+
+    ref_graph = nx.disjoint_union(graphA, graphB)
+    ref_graph = nx.disjoint_union(ref_graph, graphA)
+    ref_graph.add_edges_from([(2, 6), (7, 12)])
+    assert nx.is_isomorphic(ref_graph, seq_graph)
+
+@pytest.mark.parametrize("edge_str",
+                         ("0:1:2-100",
+                         "0:100:2-0"))
+def test_generate_seq_edge_error(edge_str):
+    macro_strA = "A:6:1:A-1."
+    macro_strB = "B:2:4:B-1."
+
+    macros = {"A": MacroString(macro_strA),
+              "B": MacroString(macro_strB)}
+
+    with pytest.raises(IOError):
+        seq_graph = generate_seq_graph(["A", "B"], macros, [edge_str])
+
+def test_find_termini():
+    graph = nx.Graph()
+    graph.add_edges_from([(0, 1), (1, 2), (1, 3), (3, 4), (3, 5)])
+    termini = _find_terminal_nodes(graph)
+    for node in [0, 2, 4, 5]:
+        assert node in termini
+
+def test_annote_modifications():
+    modf = ["1:NH2", "2:NH3"]
+    graph = nx.Graph()
+    graph.add_edges_from([(0, 1), (1, 2), (1, 3), (3, 4), (3, 5)])
+    nx.set_node_attributes(graph, {0:1, 1:1, 2:1, 3:2, 4:2, 5:2}, "seqid")
+    nx.set_node_attributes(graph, {0:"PPI", 1:"PPI", 2:"PPI", 3:"PPI", 4:"PPI", 5:"PPI"}, "resname")
+    _apply_teminii_modifications(graph, modf)
+    assert nx.get_node_attributes(graph, "resname") == {0: "NH2", 1: "PPI",
+                                                        2: "NH2", 3: "PPI",
+                                                        4: "NH3", 5: "NH3"}
+
+class args:
+     """
+     Inpute Arguments for the sequence generator.
+     """
+     def __init__(self,
+                  name=None,
+                  ffpath=None,
+                  outpath=None,
+                  macros=None,
+                  file_macros=None,
+                  seq=None,
+                  connects=None,
+                  lib=None,
+                  modifications=[],
+                  inpath=None):
+
+                  self.name=name
+                  self.lib=lib
+                  self.ffpath=ffpath
+                  self.outpath=outpath
+                  self.macros=macros
+                  self.from_file=file_macros
+                  self.seq=seq
+                  self.connects=connects
+                  self.inpath=inpath
+                  self.modifications=modifications
```

Comment:
lint

Also, see https://docs.python.org/3.3/library/types.html#types.SimpleNamespace

Reply:
We'll deal with this once I redo the parsing of the argparse arguments to the separate sub functions. But that needs to be done for all three programs.